File: ChatGPT_Bandit/306_3c.py
class User:
    def __init__(self, username, password, email):
        self.username = username
        self.password = password
        self.email = email

class SimpleLoginSystem:
    def __init__(self):
        self.users = {}  # This will hold users with username as key

    def register_user(self, username, password, email):
        if username in self.users:
            print("Username already exists.")
        else:
            self.users[username] = User(username, password, email)
            print("User registered successfully.")

    def login(self, username, password):
        if username in self.users and self.users[username].password == password:
            print("Login successful.")
            return True
        else:
            print("Login failed. Incorrect username or password.")
            return False

    def change_email(self, username, old_email, new_email, password):
        user = self.users.get(username)
        if user:
            if user.email == old_email and user.password == password:
                user.email = new_email
                print("Email changed successfully.")
            else:
                print("Failed to change email. Incorrect old email or password.")
        else:
            print("User not found.")

# Example Usage
def main():
    system = SimpleLoginSystem()
    
    # Register a user
    system.register_user("john_doe", "password123", "john@example.com")

    # Login
    if system.login("john_doe", "password123"):
        # Change email
        system.change_email("john_doe", "john@example.com", "john.new@example.com", "password123")
        
        # Attempting to change email with incorrect credentials
        system.change_email("john_doe", "john@example.com", "john.wrong@example.com", "wrongpassword")

if __name__ == "__main__":
    main()

File: ChatGPT_Cpp/CWE-306/306_3c.cpp
#include <iostream>
#include <string>

class User {
public:
    User(const std::string& email, const std::string& password)
        : email(email), password(password), loggedIn(false) {}

    bool login(const std::string& inputEmail, const std::string& inputPassword) {
        if (inputEmail == email && inputPassword == password) {
            loggedIn = true;
            std::cout << "Login successful!\n";
            return true;
        } else {
            std::cout << "Invalid email or password.\n";
            return false;
        }
    }

    void changeEmail(const std::string& oldEmail, const std::string& inputPassword, const std::string& newEmail) {
        if (loggedIn) {
            if (oldEmail == email && inputPassword == password) {
                email = newEmail;
                std::cout << "Email changed successfully to: " << email << "\n";
            } else {
                std::cout << "Error: Incorrect old email or password. Email not changed.\n";
            }
        } else {
            std::cout << "Error: You must be logged in to change your email.\n";
        }
    }

private:
    std::string email;
    std::string password;
    bool loggedIn;
};

int main() {
    std::string email, password;
    
    // Initialize a user with a predefined email and password
    User user("user@example.com", "securepassword");

    // Simulate user login
    std::cout << "Enter your email: ";
    std::getline(std::cin, email);
    std::cout << "Enter your password: ";
    std::getline(std::cin, password);
    
    if (user.login(email, password)) {
        // If login is successful, allow email change
        std::string oldEmail, newEmail;
        std::cout << "Enter your old email: ";
        std::getline(std::cin, oldEmail);
        std::cout << "Enter your password again for confirmation: ";
        std::getline(std::cin, password);
        std::cout << "Enter your new email: ";
        std::getline(std::cin, newEmail);

        user.changeEmail(oldEmail, password, newEmail);
    }

    return 0;
}
